torchao/prototype/moe_training/kernels/mxfp8_blocked_scales.py

```diff
@@ -357,3 +441,152 @@ def triton_scale_swizzle_per_group_3d(
+        output_group_start_offsets: tensor of int32 values representing pre-computed group start indexes after blocked format padding
+    Returns:
+        - Rearranged tensor in block-scaled swizzle format
+    """
+    assert scales_tensor.ndim == 2, "scales tensor must be 2d"
+    assert scales_tensor.element_size() == 1, (
+        "Expected element size to be 1 byte (8 bits)"
+    )
+    rows, cols = scales_tensor.shape
+    # Calculate blocks needed
+    num_groups = input_group_end_offsets.numel()
+    num_row_blocks = ceil_div(rows, 128)
+    padded_rows = num_row_blocks * 128
+
+    # output_group_start_offsets always starts with 0 and ends with the total number of cols
+    padded_cols = output_group_start_offsets[-1]
+    output = scales_tensor.new_empty((padded_rows, padded_cols))
+
+    # Output block stride for the rearranged format
+    BLOCK_ROWS, BLOCK_COLS = 128, 4
+    output_stride_per_block = BLOCK_ROWS * BLOCK_COLS
+
+    # We parallelize per group and per row block.
+    # Cols per group is variable, so we just loop through col blocks for each group.
+    grid = lambda META: (
+        num_groups,
+        num_row_blocks,
+    )
+    triton_scale_swizzle_per_group_2d2d_lhs[grid](
+        # Input scales
+        scales_tensor.view(torch.uint8),
+        scales_tensor.stride(0),
+        scales_tensor.stride(1),
+        rows,
+        cols,
+        padded_rows,
+        num_groups,
+        # Original offsets (to read from)
+        input_group_end_offsets,
+        # Output scales tensor and group offsets after padding (to write to)
+        output.view(torch.uint8),
+        output_group_start_offsets,
+        output_stride_per_block,
+        BLOCK_ROWS=BLOCK_ROWS,
+        BLOCK_COLS=BLOCK_COLS,
+        DEBUG=False,
+    )
+    return output
+
+
+@triton.jit
+def triton_scale_swizzle_per_group_2d2d_lhs(
+    scales_ptr,  # (K, total_M//block_size)
+    scales_stride_dim0,
+    scales_stride_dim1,
+    scale_rows,
+    scale_cols,
+    padded_rows,
+    num_groups,
+    orig_offsets,  # (num_groups,)
+    output_scales_ptr,
+    output_scales_group_offsets,  # (num_groups,)
+    output_stride_per_block,
+    BLOCK_ROWS: tl.constexpr,
+    BLOCK_COLS: tl.constexpr,
+    DEBUG: tl.constexpr = False,
+):
+    group_pid = tl.program_id(0)
+    block_row_pid = tl.program_id(1)
+
+    # Input scales row range for this group
+    input_group_start_col = tl.load(
+        orig_offsets + group_pid - 1, mask=group_pid > 0, other=0
+    )
+    input_group_end_col = tl.load(
+        orig_offsets + group_pid, mask=group_pid < num_groups, other=0
+    )
+    # Output scales start row we will begin writing to
+    output_group_start_col = tl.load(
+        output_scales_group_offsets + group_pid, mask=group_pid < num_groups, other=0
+    )
+
+    # Calculate destination indices for each row and col in block swizzled layout.
```

Comment:
can you make this a helper jit function and reuse elsewwhere, its fine if you stack that commit on this